language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/LSPackageLoader.java

```diff
@@ -350,35 +360,179 @@ public List<ModuleInfo> updatePackageMap(DocumentServiceContext context) {
         return moduleInfos;
     }
 
+    public record LSListenerIndex(List<LSPackage> ballerina, List<LSPackage> ballerinax) {
+    }
+
+    public record LSPackage(@SerializedName(value = "organization") String orgName,
+                            @SerializedName(value = "name") String module, String version, List<LSListener> listeners) {
+    }
+
+    public record LSListener(String name, List<Parameter> parameters, List<Record> records) {
+    }
+
+    public record Parameter(String name, String type, String category) {
+    }
+
+    public record Record(String name, List<Field> fields) {
+    }
+
+    public record Field(String name, String type, String category) {
+    }
+
+    public Map<String, List<ServiceTemplateGenerator.IndexedListenerMetaData>> getCachedListenerMetaData() {
+        return cachedListenerMetaData;
+    }
+
+    private static String getFileChecksum(String filePath) throws Exception {
+        FileInputStream fileInputStream = new FileInputStream(filePath);
+
+        MessageDigest messageDigest = MessageDigest.getInstance("SHA-256");
+        byte[] buffer = new byte[1024];
+
+        int n = 0;
+        while ((n = fileInputStream.read(buffer)) != -1) {
+            messageDigest.update(buffer, 0, n);
+        }
+
+        byte[] checksumBytes = messageDigest.digest();
+        StringBuilder checksum = new StringBuilder();
+        for (byte b : checksumBytes) {
+            checksum.append(String.format("%02x", b & 0xff));
+        }
+        return checksum.toString();
+    }
+
+    private void loadListeners() throws Exception {
+        // Read the listener file from the ballerina user home
+        if (Files.exists(BALLERINA_USER_HOME_INDEX)) {
+            LSListenerIndex lsListenerIndex;
+            String userHomeIndexFileChecksum = getFileChecksum(BALLERINA_USER_HOME_INDEX.toString());
+            String payload = "";
+            boolean indexUpdated = false;
+            try {
+                String expectedCheckSum = this.centralPackageDescriptorLoader.getLSPackageIndexChecksum();
+                if (!userHomeIndexFileChecksum.equals(expectedCheckSum)) {
+                    // Download the file from the central and load the listeners
+                    payload = this.centralPackageDescriptorLoader.getLSPackageIndex();
+                    lsListenerIndex = new Gson().fromJson(payload, LSListenerIndex.class);
+                    indexUpdated = true;
```

Comment:
We can call `saveLSPackageIndexInBallerinaUserHome` here and we don't need `indexUpdated`.